src/Common/DatabaseAccountUtility.ts

```diff
@@ -30,6 +30,8 @@ export function getWorkloadType(): WorkloadType {
 
 export function isGlobalSecondaryIndexEnabled(): boolean {
   return (
-    !isFabric() && userContext.apiType === "SQL" && userContext.databaseAccount?.properties?.enableMaterializedViews
```

Comment:
The Fabric changes are significant and could be done independently. Can make the changes that the Fabric behavior stays as-is in this PR and we do all the Fabric changes in a subsequent PR.